### DISCUSSION TITLE
Lazer randomly crashes on polaris with GPU drivers set to compute mode.

### DISCUSSION THREAD

**Discussion by nathanielcwm** [ppy/osu#14649] [Q&A] [answered]
**Describe the bug:** 
With the AMD GPU drivers set to compute lazer seems to randomly crash. Normally the crashes occur within the first minute of the game being launched and about half the time it leads to a GPU driver timeout / crash, but the GPU driver timeout only occurs if the game managed to reach the main menu which I haven't been able to reproduce currently. I've tried reverting to a completely clean slate of lazer by deleting the osu folders from both %appdata% and %localappdata% but it doesn't appear to help.

The crashes occurring here don't appear to match the crashes that I've encountered before, where I can get to the main menu but it causes a driver timeout. Visual Studio claimed that it was due to a System.AccessViolationException caused by: https://github.com/ppy/osu/blob/28faa8c3303f487ad47d18e6e42d2a873f410f5e/osu.Game/Database/OsuDbContext.cs#L74

However I've not been able to reproduce that today, and it slipped my mind to copy the debugger & log output during that time.

**Screenshots or videos showing encountered issue:** 
https://user-images.githubusercontent.com/22572406/132215832-aa71985b-b438-48ad-af18-22cdc6355d5f.mp4

**osu!lazer version:** 
c7abda8, not sure if it's the same crash but it also occurred on 2021.828.0, 2021.820.0 & 2021.720.0 afaik, with the stable versions the logs normally didn't contain any useful info? 

**Logs:**
[database.log](https://github.com/ppy/osu/files/7115833/database.log)
[performance.log](https://github.com/ppy/osu/files/7115834/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7115835/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7115836/updater.log)

[database.log](https://github.com/ppy/osu/files/7115892/database.log)
[performance.log](https://github.com/ppy/osu/files/7115893/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7115894/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7115895/updater.log)

System:
Windows 10 19043.1165
Ryzen 5 3600
16GB RAM
RX570 (crashes occured on both 21.6.1 and 21.8.1; crashes also occured on 21.7.1 and 21.7.2 but that was with the stable client and I didn't save the logs.

**Comment by nathanielcwm** [comment]
[visualstudio.txt](https://github.com/ppy/osu/files/7115938/visualstudio.txt)

Oh whoops I forgot to attach the debugger output

**Comment by nathanielcwm** [comment]
Seems like the DDU and fresh install of 21.8.1 I did yesterday may have helped slightly?

Haven't encountered a single driver timeout since.

**Comment by bdach** [comment, marked as answer]
I don't know how we can possibly address this without owning the exact same hardware you do. And even then this is overwhelmingly likely to be a driver error.

**Reply by nathanielcwm** [reply]
Yeah a driver error does seem most likely, especially since it's AMD and you know how AMD Windows graphics drivers are with oGL.

**Reply by nathanielcwm** [reply]
Seems to no longer occur on 2021.1211 and with the 21.12.1 drivers.
